zkvm/src/types.rs

```diff
@@ -303,6 +303,15 @@ impl Value {
     }
 }
 
+impl Expression {
+    pub fn constant(a: Scalar) -> Self {
+        Expression {
+            terms: vec![(r1cs::Variable::One(), a)],
+            assignment: None,
```

Comment:
The VM types are reused by both Prover and Verifier. So we need to keep in mind whether a specific type is used by both, or only by one. Here, we need to keep the assignment.

```suggestion
            assignment: Some(ScalarWitness::Scalar(a.into())),
```

We'd also want to retain `ScalarWitness::Integer` if we compose integers and wish to send them into a rangeproof later, but we can leave this refactoring for later PR. It would involve writing Add/Neg traits for ScalarWitness etc.

Reply:
...and also changing `S: Into<Scalar>` for `S: Into<ScalarWitness>`.